### DISCUSSION TITLE
Proper Use of Flatline

### DISCUSSION THREAD

**Discussion by NexusChak** [jertel/elastalert2#927] [Q&A] [answered]
Dear all,

I would like to use flatline as a health check for a system. If the system alive, it writes a log entry to the ELK stack every 15 minutes. My idea is to use flatline with **threshold = 1**, **timeframe = 15** minutes, which means to me that ElastAlert2 will alert me once there is zero log entry has been within 15 minutes. Please kindly correct me if there is any mis-understanding. For the query frequency, I have it set to **run_every = 10 seconds** in the config.yaml. However, it turns out ElastAlert2 keep sending me a matched alert every minute for some reason. Please kindly help me to understand the proper use of a flatline alert. Thanks!

Another weird thing, every time I start this flatline rule, it starts querying from the exact same day (2022-08-12 in this case) all the way to the current time before sending those per minute alerts.

**Comment by jertel** [comment, marked as answer]
See the docs, specifically the section regarding the [realert](https://elastalert2.readthedocs.io/en/latest/ruletypes.html?highlight=realert#realert) setting.

By default, the `buffer_time` setting controls the period of history that will be searched. But several other settings can override and/or influence this. I suggest reading the docs for more information.

**Reply by NexusChak** [reply]
Thanks for your info. I have checked out the related docs you have mentioned.

However, I have the `buffer_time` set to 15 minutes in the config.yaml, which my other two frequency alerts work just fine. So I assume the config file should be effective. 

Regarding to the realert setting, it's supposed to compress multiple alert matches. But in my case, a log entry is being sent every 15 minutes which satisfy the `threshold = 1` and `timeframe = 15 minutes` (I have try 16 minutes just in case) conditions, there shouldn't be any alert match.
